openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIResponseMapper.java

```diff
@@ -335,6 +332,9 @@ private ApiResponses getAPIResponses(OperationAdaptor operationAdaptor, ApiRespo
                 if (qNode.modulePrefix().toString().trim().equals(HTTP)) {
                     String code = generateApiResponseCode(qNode.identifier().toString().trim());
                     apiResponse.description(qNode.identifier().toString().trim());
+                    if (code.startsWith("2") && !headers.isEmpty()) {
```

Comment:
since I noticed this 2xx response validation in other places, maybe we can a util function (with an improved regex validation)